easycv/models/backbones/vitdet.py

```diff
@@ -0,0 +1,849 @@
+# --------------------------------------------------------
+# BEIT: BERT Pre-Training of Image Transformers (https://arxiv.org/abs/2106.08254)
+# Github source: https://github.com/microsoft/unilm/tree/master/beit
+# Copyright (c) 2021 Microsoft
+# Licensed under The MIT License [see LICENSE for details]
+# By Hangbo Bao
+# Based on timm, mmseg, setr, xcit and swin code bases
+# https://github.com/rwightman/pytorch-image-models/tree/master/timm
+# https://github.com/fudan-zvg/SETR
+# https://github.com/facebookresearch/xcit/
+# https://github.com/microsoft/Swin-Transformer
+# https://github.com/ViTAE-Transformer/ViTDet
+# --------------------------------------------------------'
+import math
+from functools import partial
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.checkpoint as checkpoint
+from mmcv.cnn import build_norm_layer, constant_init, kaiming_init
+from mmcv.runner import get_dist_info
+from mmdet.utils import get_root_logger
+from timm.models.layers import drop_path, to_2tuple, trunc_normal_
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from easycv.utils.checkpoint import load_checkpoint
+from ..registry import BACKBONES
+from ..utils import build_conv_layer
+
+
+class DropPath(nn.Module):
```

Comment:
NIT: duplicated code with easycv/models/backbones/vit_transfomer_dynamic.py, we should resuse these modules, add a TODO to refactor vision transformers in the future

Reply:
already record todo